storage/src/main/java/tech/pegasys/teku/storage/client/ChainHead.java

```diff
@@ -61,24 +65,27 @@ public UInt64 findCommonAncestor(final ChainHead other) {
             .max(spec.getSlotsPerHistoricalRoot(slot)) // Avoid underflow
             .minus(spec.getSlotsPerHistoricalRoot(slot));
     while (slot.isGreaterThan(minSlotWithHistoricRoot)) {
-      if (getBlockRootAtSlot(slot).equals(other.getBlockRootAtSlot(slot))) {
-        return slot;
+      final Bytes32 blockRootAtSlot = getBlockRootAtSlot(slot);
+      if (blockRootAtSlot.equals(other.getBlockRootAtSlot(slot))) {
+        return new SlotAndBlockRoot(slot, blockRootAtSlot);
       }
       slot = slot.minus(1);
     }
     // Couldn't find a common ancestor in the available block roots so fallback to finalized
-    return getState()
-        .getFinalized_checkpoint()
-        .getEpochStartSlot(spec)
-        .min(other.getState().getFinalized_checkpoint().getEpochStartSlot(spec));
+    final Checkpoint finalizedCheckpoint = getState().getFinalized_checkpoint();
+    final UInt64 finalizedBlock =
+        finalizedCheckpoint
+            .getEpochStartSlot(spec)
+            .min(other.getState().getFinalized_checkpoint().getEpochStartSlot(spec));
+    return new SlotAndBlockRoot(finalizedBlock, finalizedCheckpoint.getRoot());
```

Comment:
ooo, missed this the first time but this isn't quite right.  We need to take the root from the earliest finalized checkpoint.  So this will need to be something more like (formatting guaranteed to be wrong):
```suggestion
    final Checkpoint earliestFinalizedCheckpoint = getState().getFinalized_checkpoint().getEpoch()
        .isLessThanOrEqualTo(other.getState().getFinalized_checkpoint().getEpoch())
        ? getState().getFinalized_checkpoint()
        : other.getState().getFinalized_checkpoint();
    return new SlotAndBlockRoot(earliestFinalizedCheckpoint.getEpochStartSlot(spec), earliestFinalizedCheckpoint.getRoot());
```